Updated adxl345 sensor driver to utilze devicetree configuration

whati001 opened the pull request:
Followup PR of: https://github.com/zephyrproject-rtos/zephyr/pull/66046

Updated/rewritten adxl345 sensor code to support devicetree parameters. I have tested the code with an nrf52840-dk and a adxl345 connected via I2C.

ttmut commented on dts/bindings/sensor/adi,adxl345-common.yaml:
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2023 Andreas Karner <andreaskarner@outlook.com>
```
The year should be 2024.

ttmut commented:
You may consider rewriting the commit titles to include the area of changes such as`dts: bindings: ...` and `drivers: sensor: ...`
See https://docs.zephyrproject.org/latest/contribute/guidelines.html#area-summary-of-change

whati001 committed:
dts: bindings: sensor: Extended adxl345 sensor devicetree file definition

Added sampling frequency, range, and fifo mode configuration
to adxl345 sensor devicetree file.

Signed-off-by: Andreas Karner <andreaskarner@outlook.com>

whati001 committed:
driver: sensors: adi: Add devicetree support for adxl345 sensor

Rewritten adxl345 sensor driver to accept configuration
via devicetree system.

Signed-off-by: Andreas Karner <andreaskarner@outlook.com>

ubieda commented on drivers/sensor/adi/adxl345/adxl345.h:
```diff
@@ -19,47 +20,143 @@
-#define ADXL345_MAX_FIFO_SIZE      32
+/* ADXL345 DEVICE_ID register */
+#define ADXL345_DEVICE_ID_REG   0x00
+#define ADXL345_DEVICE_ID_VALUE 0b11100101
```
This seems like a non-related change; it already was defined in:
``` c
#define ADXL345_PART_ID            0xe5
```

Moreover, Usage of binary constants is discouraged. See: #60564.
Please discard non-related changes.

ubieda reviewed (changes requested):
Please remove non-related changes

whati001 commented:
Part of this PR were implemented in https://github.com/zephyrproject-rtos/zephyr/pull/80004.
I may open a new PR for specifying the range via dt.
